Samples.Objc/Samples.ObjC/ViewControllers/AddItemViewController/AddItemViewController.m

```diff
@@ -0,0 +1,222 @@
+//  PayPalNativeCheckoutObjC
+//
+//  Created by Nguyen, The Nhat Minh on 3/9/21.
+//
+
+#import "AddItemViewController.h"
+#import "TextFieldCell.h"
+#import "QuantityView.h"
+
+@interface AddItemViewController ()
+@property (nonatomic) NSString *itemName;
+@property (nonatomic) NSString *itemPrice;
+@property (nonatomic) NSString *titleText;
+@property (nonatomic) NSString *buttonTitle;
+@property (nonatomic) UIView *containerView;
+@property (nonatomic) UILabel *titleLabel;
+@property (nonatomic) UITableView *tableView;
+@property (nonatomic) QuantityView *quantityView;
+@property (nonatomic) UIButton *saveButton;
+@property (nonatomic) NSLayoutConstraint *containerViewBottom;
+@end
+
+@implementation AddItemViewController
+
+- (instancetype)init {
+  self = [super init];
+  self.itemName = @"";
+  self.itemPrice = @"";
+  self.titleText = @"Add to cart";
+  self.buttonTitle = @"Add";
+  return self;
+}
+
+- (instancetype)initWithItem:(PPCPurchaseUnitItem *)item {
+  self = [super init];
+  self.itemName = item.name;
+  self.itemPrice = item.unitAmount.value;
+  self.titleText = @"Edit item";
+  self.buttonTitle = @"Save";
+  return self;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  [self setupUI];
+  [self setupTableView];
+  [self setupConstraints];
+  [self registerListeners];
+}
+
+- (void)registerListeners {
+  [NSNotificationCenter.defaultCenter addObserver:self selector:@selector(keyboardWillHide:) name:UIKeyboardWillHideNotification object:nil];
+  [NSNotificationCenter.defaultCenter addObserver:self selector:@selector(keyboardWillShow:) name:UIKeyboardWillShowNotification object:nil];
+}
+
+- (void)keyboardWillHide:(NSNotification *)notification {
+  self.containerViewBottom.constant = 0;
+  [UIView animateWithDuration:0.2 animations:^{
+    [self.view layoutIfNeeded];
+  }];
+}
+
+- (void)keyboardWillShow:(NSNotification *)notification {
+  if (notification.userInfo[UIKeyboardFrameEndUserInfoKey]) {
+    self.containerViewBottom.constant = -[notification.userInfo[UIKeyboardFrameEndUserInfoKey] CGRectValue].size.height;
+    [UIView animateWithDuration:0.2 animations:^{
+      [self.view layoutIfNeeded];
+    }];
+  }
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  CAShapeLayer * maskLayer = [CAShapeLayer layer];
+  maskLayer.path = [UIBezierPath bezierPathWithRoundedRect: self.containerView.bounds byRoundingCorners: UIRectCornerTopLeft | UIRectCornerTopRight cornerRadii: (CGSize){16, 16}].CGPath;
+  self.containerView.layer.mask = maskLayer;
+}
+
+- (void)setupTableView {
+  [self.tableView registerClass:[TextFieldCell class] forCellReuseIdentifier:@"TextFieldCell"];
+  self.tableView.delegate = self;
+  self.tableView.dataSource = self;
+}
+
+- (void)setButtonEnabled {
+  if ([self.itemName isEqualToString:@""] || [self.itemPrice isEqualToString:@""]) {
+    [self.saveButton setUserInteractionEnabled:false];
+    self.saveButton.backgroundColor = [UIColor systemGray5Color];
```

Comment:
Getting the following warning for this in Xcode: `'systemGray5Color' is only available on iOS 13.0 or newer`. We can either add an available tag here or use a different color to resolve.